apps/etl/extraction/sources/emdat/extract.py

```diff
@@ -0,0 +1,195 @@
+              start_year
+              start_month
+              start_day
+              end_year
+              end_month
+              end_day
+              total_deaths
+              no_injured
+              no_affected
+              no_homeless
+              total_affected
+              reconstr_dam
+              reconstr_dam_adj
+              insur_dam
+              insur_dam_adj
+              total_dam
+              total_dam_adj
+              cpi
+              admin_units
+              entry_date
+              last_update
+            }
+          }
+        }
+        """
+
+    EMDAT_URL = "https://api.emdat.be/v1"
+    HEADERS = {"Authorization": settings.EMDAT_AUTHORIZATION_KEY}
+    classification_keys = [
+        "nat-met-ext-col",
+        "nat-met-ext-hea",
+        "nat-met-ext-sev",
+        "nat-met-sto-ext",
+        "nat-met-sto-tro",
+        "nat-met-sto-san",
+        "nat-met-sto-tor",
+        "nat-hyd-flo-fla",
+        "nat-hyd-flo-flo",
+        "nat-hyd-flo-riv",
+        "nat-hyd-flo-coa",
+        "nat-hyd-flo-ice",
+        "nat-cli-dro-dro",
+        "nat-cli-wil-for",
+        "nat-cli-wil-lan",
+        "nat-cli-wil-wil",
+        "nat-cli-glo-glo",
+        "nat-geo-ear-gro",
+        "nat-geo-ear-tsu",
+        "nat-geo-vol-ash",
+        "nat-geo-vol-lah",
+        "nat-geo-vol-lav",
+        "nat-geo-vol-pyr",
+        "nat-geo-vol-vol",
+        "nat-geo-mmd-ava",
+        "nat-geo-mmd-lan",
+        "nat-geo-mmd-roc",
+        "nat-geo-mmd-sub",
+        "nat-bio-epi-bac",
+        "nat-bio-epi-vir",
+        "nat-bio-epi-par",
+        "nat-bio-inf-ins",
+        "nat-bio-inf-gra",
+        "nat-bio-inf-loc",
+    ]
+
+    # ref: https://files.emdat.be/docs/emdat_api_cookbook.pdfhttps://files.emdat.be/docs/emdat_api_cookbook.pdf
+    variables = {"limit": -1, "include_hist": True, "classif": classification_keys}
+
+    # Create new extraction object for each extraction
+    emdat_instance = ExtractionData.objects.create(
+        source=ExtractionData.Source.EMDAT,
+        status=ExtractionData.Status.PENDING,
+        source_validation_status=ExtractionData.ValidationStatus.NO_VALIDATION,
+        hazard_type=HazardType.OTHER,
+        attempt_no=0,
+        resp_code=0,
+    )
+
+    try:
+        # Get latest emdat extraction object so that we do not need to fetch historical data
+        latest_extraction = (
+            ExtractionData.objects.filter(
+                source=ExtractionData.Source.EMDAT, status=ExtractionData.Status.SUCCESS, resp_data__isnull=False
+            )
+            .exclude(source_validation_status=ExtractionData.ValidationStatus.NO_DATA)
+            .order_by("-created_at")
+            .first()
+        )
+        if latest_extraction:
+            with latest_extraction.resp_data.open() as data_file:
+                data = data_file.read()
+
+            data_json = json.loads(data)
+            if data_json["data"]["public_emdat"]:
```

Comment:
We should maybe add another JSON field in ExtractionData to store this kind of information (maybe metadata?). Then define dataclasse/schema for that field for each data source if required. Then get this information directly instead of loading raw dataset each time.

Reply:
@thenav56 I added a json field in PR:https://github.com/IFRCGo/montandon-etl/pull/129 for this functionality.